### ISSUE TITLE
Add more performance metrics

### ISSUE THREAD

**Issue by andreassot10** [CDU-data-science-team/zero-shot#3]
@asegun-cod, the accuracy score can be inflated by great model performance in predicting one or just a few classes. It would be good to add metrics that account for this. For example, Balanced Accuracy or Matthews Correlation Coefficient.

See https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics

and https://github.com/CDU-data-science-team/pxtextmining/blob/main/pxtextmining/helpers/metrics.py.

**Comment by andreassot10** [comment]
To clarify, I don't mean drop the accuracy score- just add more metrics.

**Comment by asegun-cod** [comment]
Thanks, I will do that. I am also thinking, do you it will be useful if we have the accuracy of each individual label added also? I feel these might help us see the labels that are usually well predicted and those that are not. this might be a useful metric.
what do you think @andreassot10?

**Comment by ChrisBeeley** [comment]
Definitely. Also a confusion matrix would be great

**Comment by ChrisBeeley** [comment]
E.g. https://github.com/CDU-data-science-team/experienceAnalysis/blob/main/R/plot_confusion_matrix.R

**Comment by asegun-cod** [comment]
these metrics have now been added and committed.

**Comment by asegun-cod** [comment]
these metrics have now been added and committed.

**Comment by andreassot10** [comment]
@asegun-cod , can you please direct us to specific commits and/or permalinks when making changes so that we can easily track them.

**Comment by ChrisBeeley** [comment]
Reopening. Let's close this issue once we have a permalink to some code on the main branch

**Comment by ChrisBeeley** [comment]
Could you write up the metrics and commit to the repo so we can review?

In R we would use RMarkdown, don't really know the options in Python. I've got some code I can dig out somewhere to help if you get stuck
